src/pools.ts

```diff
@@ -29,10 +38,63 @@ export class Pools
     return await this.contract.USDC()
   }
 
+  async isPaused(): Promise<boolean> {
+    return await this.contract.paused()
+  }
+
+  private _checkIsNotPaused = async () => {
+    if (await this.isPaused()) {
```

Comment:
Do not use await calls inside of an if statement, it makes the code harder to read